examples/olsconfig.yaml

```diff
@@ -1,10 +1,10 @@
 llm_providers:
-  - name: bam
+  bam:
     url: "https://bam-api.res.ibm.com"
     credentials_path: bam_api_key.txt
     models:
       - name: ibm/granite-13b-chat-v2
-  - name: openai
+  openai:
```

Comment:
For context, we are currently defining providers as a list in the config, but converting it to dict in the actual model (to be able to access provider by the key).
So this reflects the code we have to the config.